### DISCUSSION TITLE
launch.json file

### DISCUSSION THREAD

**Discussion by jsa2** [seerge/g-helper#1274] [General]
Hello!

Firstly, I'd like to express my gratitude for developing such a fantastic application. It's evident that a lot of effort and dedication went into it.

I'm currently exploring the possibility of implementing a heatmap feature for testing purposes. Specifically, I want to see if the keyboard (x13 flow 23) has modes with minimal brightness. To aid in this endeavor, it would be immensely helpful if you could provide a launch.json configuration for debugging in Visual Studio Code.

Your assistance will greatly streamline the process and ensure that I'm working in an optimal environment.

Thank you in advance, and once again, kudos for the excellent work on the app!

**Comment by seerge** [comment]
@jsa2 hello, thanks for a nice words.

I didn't really get what do you mean under "launch.json" tho ?

The only thing i have in this ? but you don't need it to compile app ...
[launchSettings.zip](https://github.com/seerge/g-helper/files/12558977/launchSettings.zip)

**Comment by jsa2** [comment]
Hello,

Thank you for your prompt response. I primarily work with Node.js, so my expertise in .NET is limited. However, I use Visual Studio Code (VSCode) for most of my development tasks. As you might know, VSCode utilizes launch.json files within the .vscode directory for debugging configurations.

I understand that many .NET developers prefer using Visual Studio, so my request might seem a bit unconventional.

My primary goal is to set a few breakpoints and understand how the heatmap feature was previously implemented.

Here's my initial guess for a launch.json configuration for .NET:

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": ".NET Core Launch (console)",
      "type": "coreclr",
      "request": "launch",
      "preLaunchTask": "build",
      "program": "${workspaceFolder}/app/bin/Debug/netcoreapp3.1/g-helper.dll",
      "args": [],
      "cwd": "${workspaceFolder}/app",
      "stopAtEntry": false,
      "console": "internalConsole",
      "justMyCode": true,
      "symbolOptions": {
        "searchPaths": ["${workspaceFolder}/app/bin/Debug/netcoreapp3.1"]
      }
    }
  ]
}

```

**Reply by seerge** [reply]
I think it would be easier if you would just install normal VS for that :) there you can open project natively and run / debug do whatever you want with it

App uses .**NET7**, not netcore3.1, and it's **WinForms** not console.

**Reply by jsa2** [reply]
Fair enough, I will take a stab first at VSCode, but if I have no luck I will definitely continue with VS. 

Once again, thx for your fantastic work on maintaining and supporting this app

Br, Joosua

**Reply by jsa2** [reply]
Just wanted to add, that was able to get the basic debugging working with following config

![image](https://github.com/seerge/g-helper/assets/58001986/a56a389b-62f2-478a-96bd-747d1ea5f476)


```json
{
    "version": "0.2.0",
    "configurations": [
      {
        "name": ".NET 7 WinForms Launch",
        "type": "coreclr",
        "request": "launch",
        "program": "${workspaceFolder}/app/bin/Debug/net7.0-windows/GHelper.dll",
        "args": [],
        "cwd": "${workspaceFolder}/app",
        "stopAtEntry": false,
        "console": "internalConsole",
        "justMyCode": true,
        "symbolOptions": {
          "searchPaths": ["${workspaceFolder}/app/bin/Debug/net7.0-windows"]
        },
      }
    ]
  }
  
```
